csharp/src/Ice/SocketStream.cs

```diff
@@ -174,7 +175,22 @@ internal virtual async ValueTask ReceiveInitializeFrameAsync(CancellationToken c
             }
             else
             {
-                // TODO: read initialize settings?
+                // Read the protocol parameters which are encoded with the binary context encoding.
+                var istr = new InputStream(data, Ice2Definitions.Encoding);
+                int dictionarySize = istr.ReadSize();
+                var parameters = new Dictionary<int, ReadOnlyMemory<byte>>(dictionarySize);
```

Comment:
Why are you creating this dictionary? Just to check for duplicates?

Reply:
Because I find it cleaner to not mix the reading and processing of the parameters. I added the check for duplicates for consistency with Quic but I can remove it if you don't like it.